### DISCUSSION TITLE
pushing string to a variable

### DISCUSSION THREAD

**Discussion by dcpri** [slint-ui/slint#590] [Q&A]
Hello. I want to send a string value from an .rs file to the ui. I assume I have to use push it but don't know where to go with it. Something like sending "Hello World" from my about.rs to the variable "message" in the ui.about.60:

about.rs

	sixtyfps::include_modules!();
	pub fn about() {
    	let ui = AppWindow::new();
    	     let message = message.clone();
    	     move |text| About::Text.push(message { "Hello World" } );
    	ui.run();
	}

ui.about.60

	export About := Rectangle {
	     Text { text: message; }
	}

This is in preparation to sending data retrieved from an SQlite database in my rs file to a form on a ui page.

**Comment by ogoffart** [comment]
You need to expose the message as a property:

```qml
export About := Rectangle {
     property <string> message;  // <-- added this
     Text { text: message; }
}
```

Then you can just call the generated setter:

```rs
sixtyfps::include_modules!();
pub fn about() {
	let ui = AppWindow::new();
        ui.set_message("Hello World".into());
	ui.run();
}
```

I hope this helps.

**Reply by dcpri** [reply]
Excellent. Thank you.
